rmgpy/solver/base.pyx

```diff
@@ -979,32 +1008,41 @@ cdef class ReactionSystem(DASx):
                     
             if filterReactions:
                 # Calculate thresholds for reactions
-                (unimolecularThresholdRateConstant,
-                 bimolecularThresholdRateConstant,
-                 trimolecularThresholdRateConstant) = self.get_threshold_rate_constants(modelSettings)
-                unimolecularThresholdVal = toleranceMoveToCore * charRate / unimolecularThresholdRateConstant
-                bimolecularThresholdVal = toleranceMoveToCore * charRate / bimolecularThresholdRateConstant
-                trimolecularThresholdVal = toleranceMoveToCore * charRate / trimolecularThresholdRateConstant
+                (kvals_uni, kvals_bi) = self.get_threshold_rate_constants()
+
+                unimolecularThresholdVal = numpy.zeros(len(kvals_uni))
+                for i in xrange(len(kvals_uni)):
```

Comment:
If `kvals_uni` is a numpy array then I think you can get away with `unimolecularThresholdVal = toleranceMoveToCore * charRate / kvals_uni`, and similar for bi and tri molecular